test/images/mount-tester/mt.go

```diff
@@ -253,18 +251,20 @@ func testFileContent(filePath string, retryDuration int) []byte {
 	retryTime := time.Second * time.Duration(retryDuration)
 	for start := time.Now(); time.Since(start) < retryTime; time.Sleep(2 * time.Second) {
 		contentBytes, err = ioutil.ReadFile(filePath)
-		if err == nil {
-			//Expected content "mount-tester new file\n", length 22
-			if len(contentBytes) == 22 {
-				break
-			} else {
-				fmt.Printf("Unexpected length of file: found %d, expected %d.Retry", len(contentBytes), 22)
+		if err != nil {
+			fmt.Printf("Error read file %s: %v, retrying\n", filePath, err)
+			continue
+		}
+		if breakOnExpectedContent {
+			if string(contentBytes) != initialContent {
+				fmt.Printf("Unexpected file: found %s, expected %s. Retry", string(contentBytes), initialContent)
+				continue
 			}
-		} else {
-			fmt.Printf("Error read file %s: %v, retry", filePath, err)
+			fmt.Printf("content of file %q: %v\n", filePath, string(contentBytes))
+			break
 		}
-
+		fmt.Printf("content of file %q: %v\n", filePath, string(contentBytes))
```

Comment:
maybe move this before the if `breakOnExpectedContent` check, and then you don't need to repeat this inside and outside the conditional?